packages/lesswrong/lib/sql/UpdateQuery.ts

```diff
@@ -0,0 +1,103 @@
+import Query, { Atom } from "./Query";
+import Table from "./Table";
+import SelectQuery from "./SelectQuery";
+
+export type UpdateOptions = Partial<{
+  limit: number,
+  returnUpdated: boolean,
+}>
+
+/**
+ * Builds a Postgres query to update some specific data in the given table.
+ * Note that upserting is handled by `InsertQuery` instead.
+ */
+class UpdateQuery<T extends DbObject> extends Query<T> {
+  constructor(
+    table: Table,
+    selector: string | MongoSelector<T>,
+    modifier: MongoModifier<T>,
+    options?: MongoUpdateOptions<T>, // TODO: What can options be?
+    updateOptions?: UpdateOptions,
+  ) {
+    if (options?.upsert) {
+      throw new Error("To create an upserting update use an InsertQuery with conflictStrategy 'upsert'");
+    }
+
+    super(table, ["UPDATE", table, "SET"]);
+    this.nameSubqueries = false;
+
+    const set: Partial<Record<keyof T, any>> = modifier.$set ?? {};
+    for (const operation of Object.keys(modifier)) {
+      switch (operation) {
+        case "$set":
+          break;
+        case "$unset":
+          for (const field of Object.keys(modifier.$unset)) {
+            set[field] = null;
+          }
+          break;
+        case "$inc":
+          for (const field of Object.keys(modifier.$inc)) {
+            set[field] = {$add: [`$${field}`, 1]};
+          }
+          break;
```

Comment:
Do you handle the `$push` case anywhere (I'm not 100% sure this is where it would need to be handled)? We do have a couple of those in the code